File: crossflow/org.eclipse.scava.crossflow.tests/py/concurrency/sleeper.py
from crossflow.concurrency.sleeper import SleeperBase

from crossflow.concurrency.result import Result


class Sleeper(SleeperBase):

    def __init__(self):
        super().__init__()

    def consumeSleepTimes(self, sleep_time: SleepTime):
        print(f"{self.workflow.name}:Sleeper:consumeSleepTimes received {sleep_time}")
        
        # Do work here
        
        results_result = Result()
        # modify your result here
        self.sendToResults(results_result)

File: crossflow/org.eclipse.scava.crossflow.tests/src/org/eclipse/scava/crossflow/tests/concurrency/Sleeper.java
package org.eclipse.scava.crossflow.tests.concurrency;

import java.util.LinkedList;
import java.util.List;

public class Sleeper extends SleeperBase {

	private int completions = 0;
	private List<String> jobIds = new LinkedList<>();

	@Override
	public Result consumeSleepTimes(SleepTime sleepTime) throws Exception {
		System.out.println("Received " + sleepTime.getJobId() + ":" + sleepTime.getSeconds());
		jobIds.add(sleepTime.getJobId());
		Thread.sleep(1000 * sleepTime.getSeconds());
		System.out.println("Finished");
		completions++;
		return null;
	}

	public List<String> getJobIds() {
		return jobIds;
	}

	public int getExecutions() {
		return completions;
	}

}
